.buildkite/pipeline-for-tags.yml

```diff
@@ -162,5 +162,5 @@ steps:
    soft_fail: true # No artifacts to download if all the bottles are already built
    commands:
    - mkdir -p "Big Sur"
-   - buildkite-agent artifact download "*bottle.tar.gz" "Big Sur/"
+   - gh release download build.tag -D "Big Sur/" -p "*.bottle.tar.gz"
```

Comment:
It would be nice to have a comment here for future reference

Reply:
> It would be nice to have a comment here for future reference

Sure I will

> But I'd like to do a test run on CI

I will try to create a buildkite pipeline for this. GitHub actions fail to build macOS bottles

Reply:
Also, in order to test this change, we need to simulate an update of the release. How would you perform it? I suppose, that we don't want to publish something that is not a correctly-building release. Maybe we can try to do that in another repo, WDYT?

Reply:
> I will try to create a buildkite pipeline for this

Makes sense, also because this issue appears on our buildkite's pipelines and not on GA

> we need to simulate an update of the release. How would you perform it? 

I think that's not necessary. We know that the bug appears when a build has failed at least once (the building script can be modified to do so) and that the issue is not with opening the PR, but rather that some bottles are missing when we do that.

So, IMO it's sufficient to make a temporary commit (in this branch or otherwise) to trigger the build (that should fail at least for one bottle on purpose) and list the bottles in the PR step (without then opening a new PR).

Let me know if you have questions about setting that up.

Reply:
@PruStephan any update on this test run?

Reply:
I still did not get how you want this change to be tested. Current changes load hashes of bottles from the gh release, so if we're going to try it, we need, well, a new release, that will provide hashes, different from what we have. The only solution for this that I see is to manually change hashes of binaries, push the commit, and trigger loading from the latest release. What do you think about it? Also, do you think the complexity of this test is worth implementing it?

Reply:
Yes, a release needs to be created, but it can be a pre-release, which can then also be deleted.

The simplest way that came to my mind about testing this was by pushing [this commit](https://github.com/serokell/tezos-packaging/commit/5401bab153836f01ed0092949ace8ca24fd72436) in a temporary branch and then push a tag to trigger [its pipeline run](https://buildkite.com/serokell/tezos-packaging/builds/4148).

You can see the changes there, but in short I tried to make as little changes as possible to the scripts and tag pipeline definition, such that:
1. it creates a release, then builds and upload one bottle, then lists all the bottles, then deletes the release
2. it is triggered by each push of the `test-695` tag (I've already re-deleted it now)
3. it uses a temporary tag/release called `test-release`
4. both bottle build steps should fail on their first run (but upload the bottle) and not fail on the second run
5. the command that you've changed hasn't been modified, so that it is tested

While I'm writing this comment the pipeline is still running, so it's possible that I've made some mistakes, but if so it should still help you as an example of the kind of test that I had in mind.

Reply:
@PruStephan it looks like my test worked.

There are a bunch of failed builds for ARM BigSur, but those are for unrelated reasons (although they help simulate more problems).

As you can see, however, [the main job](https://buildkite.com/serokell/tezos-packaging-tags/builds/4612#018a7148-9326-43ff-89b6-8d2f234893ca) failed and that's because:
> line 2: gh: command not found

I'd imagine that this is because `gh` is only available on our build machine behind a `nix` command

Reply:
Ok, I've got it. I will update the command in the test branch, and then trigger the build once more